chain/messagepool/messagepool.go

```diff
@@ -880,6 +857,73 @@ func (mp *MessagePool) HeadChange(revert []*types.TipSet, apply []*types.TipSet)
 	return merr
 }
 
+func (mp *MessagePool) runHeadChange(from *types.TipSet, to *types.TipSet, rmsgs map[address.Address]map[uint64]*types.SignedMessage) error {
```

Comment:
there is come code duplication between this method and the actual `HeadChange`, it would be nice to merge the two methods.